contract/AElf.Contracts.Election/ElectionContract_Elector.cs

```diff
@@ -0,0 +1,238 @@
+            }
+
+            State.CandidateVotes[candidatePublicKey] = candidateVotes;
+        }
+
+        private long GetVotesWeight(long votesAmount, long lockTime)
+        {
+            return lockTime.Mul(1_0000).Div(86400).Div(270).Mul(votesAmount).Add(votesAmount.Mul(2).Div(3));
+        }
+
+        private long GetEndPeriod(long lockTime)
+        {
+            var treasury = State.ProfitContract.GetScheme.Call(State.TreasuryHash.Value);
+            return lockTime.Div(State.TimeEachTerm.Value).Add(treasury.CurrentPeriod);
+        }
+
+        #endregion
+
+        #region Withdraw
+
+        /// <summary>
+        /// Withdraw a voting,recall the votes the voted by sender.
+        /// At the same time,the Shares that the voter occupied will sub form TotalShares.
+        /// and the "VOTE" token will be returned to ElectionContract;
+        /// </summary>
+        /// <param name="input">Hash</param>
+        /// <returns></returns>
+        public override Empty Withdraw(Hash input)
+        {
+            var votingRecord = State.VoteContract.GetVotingRecord.Call(input);
+
+            var actualLockedTime = Context.CurrentBlockTime.Seconds.Sub(votingRecord.VoteTimestamp.Seconds);
+            var claimedLockDays = State.LockTimeMap[input];
+            Assert(actualLockedTime >= claimedLockDays,
+                $"Still need {claimedLockDays.Sub(actualLockedTime).Div(86400)} days to unlock your token.");
+
+            // Update Elector's Votes information.
+            var voterPublicKey = Context.RecoverPublicKey().ToHex();
+            var voterVotes = State.ElectorVotes[voterPublicKey];
+            voterVotes.ActiveVotingRecordIds.Remove(input);
+            voterVotes.WithdrawnVotingRecordIds.Add(input);
+            voterVotes.ActiveVotedVotesAmount.Sub(votingRecord.Amount);
+            State.ElectorVotes[voterPublicKey] = voterVotes;
+
+            // Update Candidate's Votes information.
+            var candidateVotes = State.CandidateVotes[votingRecord.Option];
+            candidateVotes.ObtainedActiveVotingRecordIds.Remove(input);
+            candidateVotes.ObtainedWithdrawnVotingRecordIds.Add(input);
+            candidateVotes.ObtainedActiveVotedVotesAmount.Sub(votingRecord.Amount);
+            State.CandidateVotes[votingRecord.Option] = candidateVotes;
+
+            State.TokenContract.Unlock.Send(new UnlockInput
+            {
+                Address = Context.Sender,
+                Symbol = Context.Variables.NativeSymbol,
+                Amount = GetElfAmount(votingRecord.Amount),
+                LockId = input,
+                Usage = "Withdraw votes for Main Chain Miner Election."
+            });
+
+            State.TokenContract.TransferFrom.Send(new TransferFromInput
+            {
+                From = Context.Sender,
+                To = Context.Self,
+                Amount = votingRecord.Amount,
+                Symbol = ElectionContractConstants.VoteSymbol,
```

Comment:
What if I transferred my vote tokens to somebody else? In this case I won't be able to withdraw my locked tokens. And also why would somebody want to trade vote tokens? What is the purpose of the existence of these vote tokens?